straxen/rucio.py

```diff
@@ -0,0 +1,423 @@
+import socket
+import re
+from tqdm import tqdm
+import json
+from bson import json_util
+import os
+import hashlib
+import time
+from utilix import xent_collection
+import strax
+import warnings
+try:
+    from rucio.client.client import Client
+    from rucio.client.rseclient import RSEClient
+    from rucio.client.replicaclient import ReplicaClient
+    from rucio.client.downloadclient import DownloadClient
+    from rucio.client.didclient import DIDClient
+    from rucio.common.exception import DataIdentifierNotFound
+except ModuleNotFoundError:
+    warnings.warn("No installation of rucio-clients found. Cant use rucio functionality")
+
+export, __all__ = strax.exporter()
+
+
+class TooMuchDataError(Exception):
+    pass
+
+
+class DownloadError(Exception):
+    pass
+
+
+RAW_TYPES = ['raw_records', 'raw_records_nv', 'raw_records_he']
```

Comment:
_minor_ strictly speaking, these are missing a few as you don't include the '_mv' '_aqmon' etc. Perhaps we can avoid confusion by just setting it as a class attribute of the front-end RucioFrontend.heavy_types as perhaps we some day also want to add e.g. records.

Reply:
Thanks, I like the heavy_types notation much better. I think the others are okay to download, and yeah might want to add records as well. Will do